### PR TITLE
t1120.4: Test dual-hosted repo issue sync (GitHub + Gitea)

### PR DESCRIPTION
**Pull request by marcusquinn**
## Summary

Comprehensive offline test harness validating the platform abstraction layer built in t1120.1/t1120.2/t1120.3.

**100 tests** covering:
- Platform detection from HTTPS/SSH remote URLs (GitHub, Gitea, GitLab)
- Base URL extraction and repo slug parsing
- PR/MR URL generation per platform with edge cases
- Platform dispatch routing (`init_platform`, `platform_*` functions)
- Platform-agnostic lib functions (parse, compose, ref management)
- Ref management (add/fix refs, completion evidence)
- `find_closing_pr` with platform-specific URL building
- Platform auth token resolution
- Dual-hosted scenario (GitHub + Gitea platform switching)
- Edge cases (SSH with port, `.git` suffix, `compose_issue_body`)
- Adapter function existence verification (all 3 platforms x 7 ops)

All tests are offline — mocked git remotes, no network calls required.

Ref #1679

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test suite for repository synchronization across multiple Git platforms (GitHub, Gitea, GitLab). Tests validate platform detection, URL handling, reference management, and cross-platform compatibility for dual-hosted repository workflows.

### CODE REVIEW

File: .agents/scripts/test-dual-hosted-sync.sh [hunk 1]
```diff
@@ -0,0 +1,865 @@
...
+	return 0
+}
+
+verbose() {
+	[[ "$VERBOSE" == "--verbose" ]] && echo -e "  ${YELLOW}$1${NC}" || true
+	return 0
+}
+
+assert_eq() {
+	local test_name="$1" expected="$2" actual="$3"
```
[the remark this answers is not in the document]
Comment (marcusquinn) [reply .agents/scripts/test-dual-hosted-sync.sh:48]: Dismissed: this assignment uses positional parameters ($1, $2, $3), not command substitution. The style guide rule about separating local declaration from assignment applies only when command substitution is involved (e.g., `local var=$(cmd)`) where the exit code would be masked. Simple variable assignment has no exit code risk.

File: .agents/scripts/test-dual-hosted-sync.sh [hunk 2]
```diff
@@ -0,0 +1,865 @@
...
+}
+
+# =============================================================================
+# Setup: Create mock git repos with different remotes
+# =============================================================================
+
+setup_mock_repo() {
+	local repo_dir="$1" remote_url="$2"
+	mkdir -p "$repo_dir"
+	git -C "$repo_dir" init -q 2>/dev/null
```
[the remark this answers is not in the document]
Comment (marcusquinn) [reply .agents/scripts/test-dual-hosted-sync.sh]: Fixed in 81fa177. Removed `2>/dev/null` from `git init`, `remote add`, `remote set-url`, and `remote remove` in the test setup. The `-q` flag on init already suppresses normal output; stderr should remain visible for real errors.

File: .agents/scripts/test-dual-hosted-sync.sh [hunk 3]
```diff
@@ -0,0 +1,865 @@
...
+VERBOSE="false"
+DRY_RUN="false"
+FORCE_CLOSE="false"
+REPO_SLUG=""
+PLATFORM=""
+SUPERVISOR_DB=""
+
+# Source the helper — this defines detect_platform, init_platform, etc.
+# We'll override detect_platform for tests that need it.
+source "${SCRIPT_DIR}/issue-sync-helper.sh" 2>/dev/null || {
```
[the remark this answers is not in the document]
Comment (marcusquinn) [reply .agents/scripts/test-dual-hosted-sync.sh]: Fixed in 81fa177. Removed `2>/dev/null` from the source command. The `|| { fallback }` already handles failure, and syntax errors in the sourced file should be visible for debugging.

File: .agents/scripts/test-dual-hosted-sync.sh [hunk 4]
```diff
@@ -0,0 +1,865 @@
...
+
+echo ""
+echo "--- 12. Gitea Adapter Logic ---"
+
+# Test that gitea_list_issues builds correct query params
+test_gitea_list_state_all() {
+	# We can't call the real API, but we can verify the state_param logic
+	# by checking the function exists and has the right structure
+	TOTAL=$((TOTAL + 1))
+	if declare -f gitea_list_issues >/dev/null 2>&1; then
```
[the remark this answers is not in the document]
Comment (marcusquinn) [reply .agents/scripts/test-dual-hosted-sync.sh]: Fixed in 81fa177. Removed `2>&1` from all `declare -f` checks (lines 791, 816, 834). `declare -f` only writes to stdout, never stderr, so `>/dev/null` alone is sufficient.
